### PR TITLE
[lint] Actually make flake8 do something

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30892 [lint] Actually make flake8 do something**

Fixes all outstanding lints and actually installs a properly configured
flake8

Differential Revision: [D18862825](https://our.internmc.facebook.com/intern/diff/D18862825)

### CODE REVIEW

File: test/dist_autograd_test.py [hunk 1]
```diff
@@ -1168,6 +1168,7 @@ def test_backward_simple_python_udf(self):
                 loss = ret.sum()
                 local_grads = self._verify_backwards(exec_mode, [loss], context_id, local_grads, t1, t2)
 
+    @staticmethod
```
Comment (albanD) [inline test/dist_autograd_test.py]: How were these called before?

File: torch/jit/__init__.py [hunk 2]
```diff
@@ -1741,7 +1741,7 @@ def __contains__(self, key):
         # dir is defined by the base nn.Module, so instead of throwing if
         # it is not overriden, we call into the nn.Module __dir__ method
         def __dir__(self):
-            self_method = getattr(self, "__dir__")
+            self_method = self.__dir__
```
Comment (albanD) [inline torch/jit/__init__.py]: Was there a good reason to do this? Why was getattr used to begin with?

**Review by albanD — APPROVED** [review]
Looks good, small questions.
Not sure if the fbgemm is expected though.

File: test/dist_autograd_test.py (hunk 1 above)
Comment (suo) [reply test/dist_autograd_test.py]: they are passed as function objects to the rpc system, which I think didn't treat them as methods (coincidentally)

File: torch/jit/__init__.py (hunk 2 above)
Comment (suo) [reply torch/jit/__init__.py]: I don't think so? They are basically the same when used with a string literal, so I guess it was just reflex.

File: test/dist_autograd_test.py (hunk 1 above)
Comment (albanD) [reply test/dist_autograd_test.py]: Ho ok... nice catch then :)

File: torch/jit/__init__.py (hunk 2 above)
Comment (albanD) [reply torch/jit/__init__.py]: Ok, I'm not sure why you would ever use `getattr` with constant strings. That's why I was asking.
But the fix looks better yes.

**Comment by kostmo** [comment]
## CircleCI build failures summary
As of commit c595231:
* **1/1** broken upstream at merge base 26c5146 (see [grid view](https://dr.pytorch.org/master-timeline.html?count=10&sha1=26c51468c5572869743346edaf017e00e93271b2&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis (WIP)

Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=c5952312eb50cbf26a4f4c7d65ec80004b0d3fd5):

### 1 failure recognized by patterns:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/3832856) pytorch_linux_xenial_py3_clang5_asan_build
**Step:** Build
```
Dec 06 21:34:04 caused by: Connection refused (os error 111)
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

**Pull request closed without merging**